pkg/state/state_nic_feature_discovery_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+2024 NVIDIA CORPORATION & AFFILIATES
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	. "github.com/onsi/gomega"
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	mellanoxv1alpha1 "github.com/Mellanox/network-operator/api/v1alpha1"
+	"github.com/Mellanox/network-operator/pkg/render"
+	"github.com/Mellanox/network-operator/pkg/utils"
+)
+
+func Test_stateNICFeatureDiscovery_GetManifestObjects(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	logger := log.FromContext(ctx)
+	manifestDir := "../../manifests/state-nic-feature-discovery"
+	files, err := utils.GetFilesWithSuffix(manifestDir, render.ManifestFileSuffix...)
+	g.Expect(err).ToNot(HaveOccurred())
+	renderer := render.NewRenderer(files)
+	skel := stateSkel{client: fake.NewClientBuilder().Build(), renderer: renderer}
+
+	cr := nicClusterPolicyWithBaseFields.DeepCopy()
+	cr.Spec.NicFeatureDiscovery = &mellanoxv1alpha1.NICFeatureDiscoverySpec{
+		ImageSpec: mellanoxv1alpha1.ImageSpec{
+			Image:            "image-one",
+			Repository:       "repository",
+			Version:          "five",
+			ImagePullSecrets: []string{"secret-one", "secret-two"},
+			ContainerResources: []mellanoxv1alpha1.ResourceRequirements{
+				{
+					Name:   "first-resource",
+					Limits: map[corev1.ResourceName]resource.Quantity{},
+					Requests: map[corev1.ResourceName]resource.Quantity{
+						"resource-one": {Format: "format-one"}},
+				},
+			},
+		},
+	}
+	t.Run("test GetObjectManifests", func(t *testing.T) {
```

Comment:
Let's try to keep same style as other tests (Context, It...)

Reply:
Can I ask why we use ginkgo for these unit tests? I'm fine to keep the same style if we have to - but it looks like we don't use any of the features of ginkgo in our tests really - and it makes them much more difficult to run locally for me.

Reply:
main point is consistency across many projects which we are working on.

other points:
- common init /cleanup of variables for group of tests
- ability to describe what actually a test is testing
- table tests will look the same (instead of re-implementing crude mechanism as i repeatedly saw in other proj)
- focus on subset tests for test development / debug
- test randomizaiton (yea i dont think we really need it but its there ...)
- later on if needed, can leverage other features

Reply:
Fair enough on the main point - I'm not sure I agree with any of the other points compared to the go test framework, but it's not really important if we're trying to keep consistent.